python/cudf/cudf/io/json.py

```diff
@@ -21,13 +21,27 @@ def read_json(
     lines=False,
     compression="infer",
     byte_range=None,
+    keep_quotes=False,
     *args,
     **kwargs,
 ):
     """{docstring}"""
 
-    if engine == "cudf" and not lines:
-        raise ValueError("cudf engine only supports JSON Lines format")
+    if not isinstance(dtype, (abc.Mapping, bool)):
+        warnings.warn(
+            "passing 'dtype' as list is deprecated, instead pass "
+            "a dict of column name and types key-value paris."
+            "in future versions 'dtype' can only be a dict or bool",
+            FutureWarning,
+        )
+
+    if engine in {"cudf", "cudf_experimental"} and not lines:
```

Comment:
With PR #11714, `cudf_experimental` will support both JSON lines, and also records.
Also, old `cudf_experimental` also supports records format.
```suggestion
    if engine  == "cudf" and not lines:
```

Reply:
@karthikeyann, is #11714 ready to merge for 22.10?

Reply:
if tests pass, it should be. Addressed most of review comments, left some for next PRs.
I merged this PR too locally and tested with above change. Tests pass. 
experimental engine could use more python tests.

Reply:
is `"[1, 2, 3]\n[4, 5, 6]\n[7, 8, 9]\n"` a valid JSON lines?
This is not supported by `cudf_experimental` engine. Record orient with lines=True/False is only supported.

Reply:
> is `"[1, 2, 3]\n[4, 5, 6]\n[7, 8, 9]\n"` a valid JSON lines?

It is:
```python
In [1]: import pandas as pd

In [2]: json = "[1, 2, 3]\n[4, 5, 6]\n[7, 8, 9]\n"

In [5]: pd.read_json(json, lines=True, orient="records")
Out[5]: 
   0  1  2
0  1  2  3
1  4  5  6
2  7  8  9
```